x/gravity/keeper/keeper.go

```diff
@@ -220,24 +240,46 @@ func (k Keeper) DeserializeValidatorIterator(vals []byte) stakingtypes.ValAddres
 // it is invalid for a subset of ERC20 addresses. (2) is not yet implemented
 // Blocking some addresses is technically motivated, if any ERC20 transfers in a batch fail the entire batch
 // becomes impossible to execute.
-func (k Keeper) InvalidSendToEthAddress(ctx sdk.Context, addr types.EthAddress, _erc20Addr types.EthAddress) bool {
+func (k Keeper) InvalidSendToEthAddress(ctx context.Context, addr types.EthAddress, _erc20Addr types.EthAddress) bool {
 	return addr == types.ZeroAddress()
 }
 
-func (k Keeper) GetStore(ctx sdk.Context) sdk.KVStore {
-	return k.storeGetter.Store(ctx)
+func (k Keeper) GetStore(ctx context.Context) storetypes.KVStore {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	return k.storeGetter.Store(sdkCtx)
 }
 
 type GravityStoreGetter struct {
 	storeKey storetypes.StoreKey
 }
 
+func (gsg GravityStoreGetter) GetChainInfo(ctx context.Context, targetChainReferenceID string) (*evmmoduletypes.ChainInfo, error) {
```

Comment:
Why we need these functions?

Reply:
Its being used in 3 places
[batch.go](https://github.com/palomachain/paloma/blob/2226e95be60c4972dff44afbbbe7b4cb0ec268c8/x/gravity/keeper/batch.go#L39)
[evidence.go](https://github.com/palomachain/paloma/blob/2226e95be60c4972dff44afbbbe7b4cb0ec268c8/x/gravity/keeper/evidence.go#L39)
[msg_server.go](https://github.com/palomachain/paloma/blob/2226e95be60c4972dff44afbbbe7b4cb0ec268c8/x/gravity/keeper/msg_server.go#L80)

Reply:
removed ununsed functions and pushed